File: python/example/selenium/Pages/account/login/login_page.py
from selenium.webdriver.common.by import By
from citronella import ui
from Pages.components import Header, Footer


class LoginPage(Header, Footer):

    def username_input(self):
        return ui(By.ID, 'username')

    def password_input(self):
        return ui(By.ID, 'password')

    def login_button(self):
        return ui(By.XPATH, '//input[@type="submit"]')

File: javascript/example/page/login/login-page.js
const { By } = require("selenium-webdriver");
const { ui } = require("citronella");

class LoginPage {
  get usernameInput() {
    return ui(By.id("login_username"));
  }

  get passwordInput() {
    return ui(By.id("login_password"));
  }

  get signinButton() {
    return ui(By.css('button[type="submit"]'));
  }

  get createAccountLink() {
    return ui(By.css('a[href="/signup"'));
  }
}

module.exports = LoginPage;
